pkg/controller/controller_broker.go

```diff
@@ -336,14 +335,25 @@ func (c *controller) reconcileClusterServiceBroker(broker *v1alpha1.ClusterServi
 			return stderrors.New(s)
 		}
 
-		for _, servicePlan := range servicePlans {
-			glog.V(4).Infof("ClusterServiceBroker %q: reconciling ClusterServicePlan (K8S: %q ExternalName: %q)", broker.Name, servicePlan.Name, servicePlan.Spec.ExternalName)
-			if err := c.reconcileClusterServicePlanFromClusterServiceBrokerCatalog(broker, servicePlan); err != nil {
+		existingServiceClasses, existingServicePlans, err := c.getCurrentServiceClassesAndPlansForBroker(broker)
+		if err != nil {
+			return err
+		}
+
+		existingServiceClassMap := convertServiceClassListToMap(existingServiceClasses)
```

Comment:
It might be useful to put a comment here stating the high level approach:
 - create maps for classes/plans that currently exist in the SC
 - create maps for classes/plans that currently exist on the Broker
 - reconcile all the classes/plans that currently exist on the Broker. This will create any new Classes/Plans, and update existing ones
 - remove all the plans/classes from the maps that are on the broker maps
 - remaining plans/classes in the maps are plans/classes that need to be removed.

Reply:
yeah, totally agree - and was in my plan for 'finishing touches' on this.

Reply:
Inlined comments are fine, I was actually thinking of having one place on the top before starting so that you can at a glance see what this is supposed to do.